### ISSUE TITLE
Is this wrong warning?

### ISSUE THREAD

**Issue by vy0b0x** [rust-lang/rust#45128]
```
fn main() {
    let mut done: bool;
    {
        done = false;
    }
    if done {
        println!("hoho");
    } else {
        println!("blabla");
    }
}
```
```
warning: variable does not need to be mutable
 --> src/main.rs:2:9
  |
2 |     let mut done: bool;
  |         ^^^^^^^^
  |
  = note: #[warn(unused_mut)] on by default

```
rustc 1.20.0 (f3d6973f4 2017-08-27)
cargo 0.21.0 (5b4b8b2ae 2017-08-12)
rustup 1.6.0 (a11c01e8c 2017-08-30)

**Comment by sinkuu** [comment]
No, the variable is initialized but not mutated.

https://doc.rust-lang.org/book/first-edition/variable-bindings.html#initializing-bindings

**Comment by vy0b0x** [comment]
Thanks
